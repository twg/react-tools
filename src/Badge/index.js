import React from 'react'
import css from './style.css'
import PropTypes from 'prop-types'

const getClasses = props => {
  const classes = [css.badge]
  if (props.rounded) classes.push(css.rounded)
  if (props.secondary) classes.push(css.secondary)
  if (props.caution) classes.push(css.caution)
  if (props.error) classes.push(css.error)
  return classes.join(' ')
}

export const Badge = props => (
  <div className={getClasses(props)}>{props.label}</div>
)

Badge.propTypes = {
  /**
   * Determines whether the borders of the card should be rounded
   */
  rounded: PropTypes.bool,
  /**
   * Value to display inside the badge
   */
  label: PropTypes.string,
  /**
   * A modifier to change styling to be a secondary badge
   */
  secondary: PropTypes.bool,
  /**
   * A modifier to change styling to be a caution badge
   */
  caution: PropTypes.bool,
  /**
   * A modifier to change styling to be an error badge
   */
  error: PropTypes.bool
}

export default Badge
