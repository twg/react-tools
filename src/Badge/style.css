.badge {
    display: inline-block;
    background: rgb(51, 51, 51);
    color:white;
    padding: 5px;
    font-family: Arial, Helvetica, sans-serif;
}

.rounded{
    border-radius: 15px;
}

.secondary {
    background: rgb(0, 255, 221);
    color:black;
}

.caution {
    background: rgb(247, 114, 5);
    color: black;
}

.error {
    background: rgb(255, 60, 0);
}