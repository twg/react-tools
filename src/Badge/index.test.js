/* eslint-env jest */

import React from 'react'
import Badge from './'
import renderer from 'react-test-renderer'
import cases from 'jest-in-case'

cases(
  'matches snapshots',
  opts => {
    expect(
      renderer
        .create(
          <div style={{ width: '400px' }}>
            <Badge {...opts.props} />
          </div>
        )
        .toJSON()
    ).toMatchSnapshot()
  },
  [
    { name: 'normal', props: { label: 'Wow!' } },
    { name: 'rounded', props: { label: 'Wow!', rounded: true } },
    { name: 'secondary', props: { label: 'Wow!', secondary: true } },
    { name: 'caution', props: { label: 'Wow!', caution: true } },
    { name: 'error', props: { label: 'Wow!', error: true } }
  ]
)
