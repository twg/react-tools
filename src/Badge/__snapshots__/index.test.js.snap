// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots caution 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div
    className="badge caution"
  >
    Wow!
  </div>
</div>
`;

exports[`matches snapshots error 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div
    className="badge error"
  >
    Wow!
  </div>
</div>
`;

exports[`matches snapshots normal 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div
    className="badge"
  >
    Wow!
  </div>
</div>
`;

exports[`matches snapshots rounded 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div
    className="badge rounded"
  >
    Wow!
  </div>
</div>
`;

exports[`matches snapshots secondary 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <div
    className="badge secondary"
  >
    Wow!
  </div>
</div>
`;
